core/src/openid.cpp

```diff
@@ -530,7 +531,15 @@ namespace irods::http::openid
 			// Manually verify 'typ' matches what is specified in OJWT
 			// Allow for 'JWT'. Typical pre OJWT use had such claims, based on the JWT standard.
 			// See OJWT Section 4
-			if (!(token_type == "at+jwt" || token_type == "application/at+jwt" || token_type == "jwt")) {
+			if (_type == TokenType::Access &&
+			    !(token_type == "at+jwt" || token_type == "application/at+jwt" || token_type == "jwt"))
+			{
+				logging::error("{}: Access Token with [typ] of type [{}] is not supported.", __func__, token_type);
+				return std::nullopt;
+			}
+			// If we are validating an ID Token, we only care that the 'typ' is set to 'jwt',
+			// based on the JWT standard.
+			if (_type == TokenType::ID && token_type != "jwt") {
 				logging::error("{}: Access Token with [typ] of type [{}] is not supported.", __func__, token_type);
```

Comment:
Should this say "ID Token" or "Nonstandard Access Token" instead of "Access Token"? Or are all of them "Access Token"s?

Reply:
Updated.